src/compiler/binder.ts

```diff
@@ -403,14 +405,19 @@ namespace ts {
                                 if (symbol.declarations && symbol.declarations.length &&
                                     (node.kind === SyntaxKind.ExportAssignment && !(<ExportAssignment>node).isExportEquals)) {
                                     message = Diagnostics.A_module_cannot_have_multiple_default_exports;
+                                    multipleDefaultExports = true;
                                 }
                             }
                         }
 
                         forEach(symbol.declarations, declaration => {
                             file.bindDiagnostics.push(createDiagnosticForNode(getNameOfDeclaration(declaration) || declaration, message, getDisplayName(declaration)));
                         });
-                        file.bindDiagnostics.push(createDiagnosticForNode(getNameOfDeclaration(node) || node, message, getDisplayName(node)));
+                        const declarationName = getNameOfDeclaration(node) || node;
+                        const diag = createDiagnosticForNode(declarationName, message, getDisplayName(node))
+                        file.bindDiagnostics.push(
+                            multipleDefaultExports ? addRelatedInfo(diag, createDiagnosticForNode(declarationName, Diagnostics.This_export_conflicts_with_the_first)) : diag

```

Comment:
Both the related info and the original diagnsotic have the same span here - seems wrong. The related span should be the first declaration, not this declaration.